src/Components/PriceRange/Utils/parsePriceRange.ts

```diff
@@ -1,8 +1,57 @@
-import { DEFAULT_PRICE_RANGE } from "Components/PriceRange/constants"
+import {
+  CustomRange,
+  DEFAULT_CUSTOM_RANGE,
+  DEFAULT_PRICE_RANGE,
+  PRICE_RANGE_FORMAT,
+} from "Components/PriceRange/constants"
 
-export const parsePriceRange = (range: string = DEFAULT_PRICE_RANGE) => {
-  return range.split("-").map(s => {
-    if (s === "*") return s
-    return parseInt(s, 10)
-  })
+const parsePart = (part: string): number | "*" => {
+  if (part === "*") return part
+  const parsed = parseFloat(part)
+  return isNaN(parsed) ? "*" : Math.floor(parsed)
+}
+
+const handleInvalidFormat = (range: string): CustomRange => {
+  const parts = range.split("-")
+
+  if (parts.length === 2) {
+    const lower = parts[0] ? parsePart(parts[0]) : "*"
+    const upper = parts[1] ? parsePart(parts[1]) : "*"
+
+    return [lower, upper]
+  }
+
+  return DEFAULT_CUSTOM_RANGE
+}
+
+export const parsePriceRange = (
+  range: string = DEFAULT_PRICE_RANGE
+): CustomRange => {
+  const match = range.match(PRICE_RANGE_FORMAT)
+
+  if (match) {
+    const lower = parsePart(match[1])
+    const upper = parsePart(match[3])
+
+    return [lower, upper]
+  }
+
+  return handleInvalidFormat(range)
+}
+
+export const priceRangeToLabel = (range: string) => {
+  const [min, max] = parsePriceRange(range)
+
+  if (min === "*" && max === "*") {
+    return "$0+"
+  }
+
+  if (min === "*") {
+    return `$0–$${max}`
+  }
+
+  if (max === "*") {
+    return `$${min}+`
+  }
+  return `$${min}–$${max}`
```

Comment:
Can we add some tests for these utils?

Reply:
There's tests for the others. I just pulled this one out of a the filters context. Might as well.

Reply:
Yah seems easiest enough to just chatgpt it real quick 👍